external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -586,6 +592,29 @@ fn top_level_address_(
     }
 }
 
+fn top_level_address_opt(context: &mut DefnContext, ln: P::LeadingNameAccess) -> Option<Address> {
+    let name_res = check_valid_address_name(context, &ln);
+    let named_address_mapping = context.named_address_mapping.as_ref().unwrap();
+    let sp!(loc, ln_) = ln;
+    match ln_ {
+        P::LeadingNameAccess_::AnonymousAddress(bytes) => {
+            debug_assert!(name_res.is_ok());
+            Some(Address::anonymous(loc, bytes))
+        }
+        P::LeadingNameAccess_::GlobalAddress(name) => {
+            context.env.add_diag(diag!(
+                Syntax::InvalidAddress,
+                (loc, "Top-level addresses cannot start with '::'")
+            ));
```

Comment:
Bit confused, this error seems new? But I don't see it tripped anywhere

Reply:
This was copy/pasted from from when it was being treated as such in `top_level_address_`, though looking at the code I believe both cases should be unreachable. I converted them to produce `ice`s.